web/concrete/src/File/Image/BasicThumbnailer.php

```diff
@@ -15,6 +14,39 @@ class BasicThumbnailer
     protected $jpegCompression;
 
     /**
+     * @var StorageLocation
+     */
+    private $storageLocation;
+
+    /**
+     * BasicThumbnailer constructor.
+     * Set the storage location to the the default, if it exists.
+     */
+    public function __construct()
+    {
+        $sl = StorageLocation::getDefault();
+        if(is_object($sl)) {
+            $this->setStorageLocation($sl);
+        }
```

Comment:
This is better served as part of the getStorageLocation method,

```php
if ($this->storageLocation === null) {
    $this->setStorageLocation(StorageLocation::getDefault());
}
```

`StorageLocation::getDefault()` shouldn't need to run when a storagelocation is provided.